src/metpy/plots/skewt.py

```diff
@@ -373,6 +373,38 @@ def plot(self, p, t, *args, **kwargs):
         t, p = _delete_masked_points(t, p)
         return self.ax.plot(t, p, *args, **kwargs)
 
+    def _get_line_starting_temps(self, t0=None, spacing=10):
+        r"""Determine starting temperatures for adiabats etc.
+
+        Return starting teperatures for subroutine lines, based on xaxis
+
+        Parameters
+        ----------
+        t0 : array_like, optional
+            range of temperatures to be converted to match the xaxis
+        spacing : int or float, optional
+            Spacing between t0 values, default is 10
+
+        Returns
+        -------
+        units.Quantity
+        """
+        # set default temperatures
+        if t0 is None:
+            # minimum and maximum temperature
+            xmin, xmax = self.ax.get_xlim()
+            t0 = np.arange(xmin, xmax + 1, spacing) * self.ax.xaxis.units
+
+        # if temperature has no units, guess from the lower bound
+        if not hasattr(t0, 'units'):
+            if np.nanmin(t0) > 150:
+                t0 = t0 * units.degK
+            else:
+                t0 = t0 * units.degC
```

Comment:
Do we need this empirical check, now that units are the default in `SkewT`? If we do, we need a test to hit this section.

Reply:
My lean would be that we don't need it, but can be convinced otherwise...

Reply:
It is just in case someone calls the method with their own t0, without setting t0 units. I would prefer to enforce t0 has units when calling the skewt plot_adiabat subroutines but I don't know how to use decorators.
I could add a test case for coverage or else I could remove it if someone can show me how to impose unit constraints on the optional argument t0.

Reply:
My two cents: it's a private function, so we don't need to worry about an end-user messing things up here - they use this function at their own risk. The public API that would feed into this would always have units unless the end user explicitly stripped units out of the axes.

Reply:
Ah yes sorry, I mean that if someone calls skewt.plot_dry_adiabat(t0=np.arange(250,301,10)) then the unitless t0 gets passed to the private function, which may or may not match the x axis units (xaxis is degC even when first call to skew.plot() uses kelvin for the t profile input).

Reply:
Ok, you convinced me. :wink: